src/App.tsx

```diff
@@ -15,36 +15,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_ALPHABET = 'aplha';
+const SORT_FIELD_LENGTH = 'length';
+
+type Field = '' | typeof SORT_FIELD_ALPHABET | typeof SORT_FIELD_LENGTH;
+
+function getPrepearedGoods(
+  goods: string[],
+  sortField: Field,
+  isReversed: boolean,
+): string[] {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABET:
+        prepearedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SORT_FIELD_LENGTH:
+        prepearedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        return prepearedGoods;
+    }
+  }
+
+  if (isReversed) {
+    return prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<Field>('');
+  const [isReversed, setIsReversed] = useState<boolean>(false);
```

Comment:
it's redundant to specify types for primitives 
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```